### PR TITLE
add more feature view fields

### PR DESCRIPTION
**Pull request by ddl-joyce-zhao**
## Description
Added more Feature view fields

## Related Issue

https://dominodatalab.atlassian.net/browse/DOM-43536

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [x] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/dominodatalab/domino-data/blob/main/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: openapi/featurestore.yaml [hunk 1]
```diff
@@ -404,11 +415,14 @@ components:
...
           additionalProperties:
             type: string
+        description:
+          type: string
       required:
         - name
         - features
         - entities
         - tags
+        - description
```
Comment (ddl-catherinetu) [inline openapi/featurestore.yaml:425]: for description being required, are we going to pass in an empty one if the user doesn't define one?

File: domino_data/_feature_store/sync.py [hunk 2]
```diff
@@ -110,14 +111,23 @@ def update_feature_views(commit_id: str, repo_path: str) -> None:
...
                 for x, y in zip(fv.entities, fv.entity_columns)
             ],
-            features=[Feature(name=f.name, dtype=str(f.dtype)) for f in fv.features],
+            features=[
+                Feature(name=f.name, dtype=str(f.dtype), tags=FeatureTags.from_dict(f.tags))
+                for f in fv.features
+            ],
             ttl=UNSET if fv.ttl is None else int(fv.ttl.total_seconds() * 1000),
             tags=FeatureViewRequestTags.from_dict(fv.tags),
+            description=fv.description,
```
Comment (ddl-catherinetu) [inline domino_data/_feature_store/sync.py:120]: description isn't required field when making feature view, what if fv doesn't have a description?

File: openapi/featurestore.yaml (hunk 1 above)
Comment (ddl-joyce-zhao) [reply openapi/featurestore.yaml:425]: I checked feast FeatureView implementation, although it's optional when defining a FeatureView, feast will set the value to be an empty string if not provided. So when we are reading a FeatureView object, that field is always there.

File: domino_data/_feature_store/sync.py (hunk 2 above)
Comment (ddl-joyce-zhao) [reply domino_data/_feature_store/sync.py:120]: See my above comments. This field is am empty string if not provided when initializing the feature view.

File: domino_data/_feature_store/client.py [hunk 3]
```diff
@@ -44,17 +44,30 @@ def __attrs_post_init__(self):
...
 
         Raises:
             ServerException: if update fails
         """
-        request = UpsertFeatureViewsRequest(feature_views=feature_views, git_commit_hash=commit_id)
+        request = UpsertFeatureViewsRequest(
+            feature_views=feature_views,
+            git_commit_hash=commit_id,
+            project_id=project_id,
+            user_id=user_id,
```
Comment (ddl-gabrielhaim) [inline domino_data/_feature_store/client.py:69]: should this come from the principal? or are we ok with it being passed by the client

**Review by ddl-gabrielhaim — APPROVED** [review]

**Pull request merged**